core/server/api/settings.js

```diff
@@ -228,6 +228,11 @@ settings = {
                 // We convert this to JSON, by calling toJSON on each Model (using invokeMap for ease)
                 // We use keyBy to create an object that uses the 'key' as a key for each setting.
                 var settingsKeyedJSON = _.keyBy(_.invokeMap(settingsModelsArray, 'toJSON'), 'key');
+                // Reload translations for theme when saving group of settings that includes language.
+                // There is a previous call to i18n.init() to reload the blos object when changing the
+                // language of the theme, needed to prevent side effects from language conflicts.
+                i18n.init();
+                i18n.loadThemeTranslations();
```

Comment:
This is still called too often.

e.g. reset your db and serve to /ghost.

> [2017-12-12 21:05:20] WARN Theme's file locales/en.json not found.
> INFO [2017-12-12 21:05:20] "GET /ghost/" 200 364ms
> [2017-12-12 21:05:20] WARN Theme's file locales/en.json not found.

I would suggest to listen on the model event if the settings language get's changed. By this you decouple the logic.

Reply:
That extra call to core `i18n.init()` when changing theme language (to prevent language conflicts in some situations) is no longer needed with today's commit _[i18n] Separate `coreStrings` and `themeStrings` objects_, and has been removed.

Reply:
Also, as suggested, added an event listener to reload theme translations when locale changes, instead of when saving the group of settings that includes language.